test/components/icon_button_test.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class PrimerIconButtonTest < Minitest::Test
+  include Primer::ComponentTestHelpers
+
+  def test_renders
+    render_inline(Primer::IconButton.new(icon: :star, "aria-label": "Label"))
+
+    assert_selector("button[aria-label='Label'].btn-octicon") do
+      assert_selector(".octicon.octicon-star")
+    end
+  end
+
+  def test_does_not_render_content
+    render_inline(Primer::IconButton.new(icon: :star, "aria-label": "Label")) { "content" }
+
+    refute_text("content")
```

Comment:
Do you think we should provide stronger feedback (such as an error) if content is passed to components we don't want to accept it?

Reply:
we can come up with something in a later PR!